app/src/lib/stores/repository-state-cache.ts

```diff
@@ -0,0 +1,142 @@
+import { Branch } from '../../models/branch'
+import { Commit } from '../../models/commit'
+import { PullRequest } from '../../models/pull-request'
+import { Repository } from '../../models/repository'
+import {
+  CommittedFileChange,
+  WorkingDirectoryFileChange,
+  WorkingDirectoryStatus,
+} from '../../models/status'
+import { TipState } from '../../models/tip'
+import {
+  ComparisonView,
+  IBranchesState,
+  IChangesState,
+  ICompareState,
+  IRepositoryState,
+  RepositorySectionTab,
+} from '../app-state'
+import { ComparisonCache } from '../comparison-cache'
+import { IGitHubUser } from '../databases'
+import { merge } from '../merge'
+import { GitHubUserStore } from './github-user-store'
+
+export class RepositoryStateCache {
+  private repositoryState = new Map<string, IRepositoryState>()
+
+  public constructor(private gitHubUserStore: GitHubUserStore) {}
```

Comment:
Could we pass in a function that returns a `GitHubUserStore` instead of keeping this as a dependency?

Reply:
Also, I only spotted one usage of this store on line 34, maybe we can just pass a callback that make the call to `getUsersForRepository`, that should make testing a bit easier.

Reply:
Maybe? I hadn't really thought about making that dependency more abstract. I'll have a play around and see how it feels.

Reply:
Try out https://github.com/desktop/desktop/pull/5564/commits/22c2428c7393e50dc9a4dd491abca14f5b26168b and see how it feels